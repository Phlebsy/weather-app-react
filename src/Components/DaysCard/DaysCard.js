import React from 'react';
import PropTypes from 'prop-types';
import { convertFromKelvin, getIcon } from '../../utilities/utilities';

// DaysCard represents the aggregate data for each individual day
// Logic is passed in from parent component, in order to change the active day
export default function DaysCard(props) {
  let { day, onClick } = props;
  return (
    <div
      onClick={onClick}
      className="d-flex flex-column align-items-center py-3 bg-success"
    >
      <i className={'fas fa-' + getIcon(undefined, day)} />
      <p className="m-0">{convertFromKelvin(day['avg'])}&deg;</p>
      <p className="m-0">{day['fullDay']}</p>
    </div>
  );
}

DaysCard.propTypes = {
  day: PropTypes.object.isRequired,
  onClick: PropTypes.func
};
