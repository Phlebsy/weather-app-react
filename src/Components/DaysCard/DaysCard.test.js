import React from 'react';
import ReactDOM from 'react-dom';
import DaysCard from './DaysCard';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const dayObj = {
  avg: 276.25,
  fullDay: 'Tue - Dec 4',
  weather: 'clear sky'
};

describe('<DaysCard />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DaysCard day={dayObj} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders icon correctly', () => {
    const wrapper = shallow(<DaysCard day={dayObj} />);
    expect(wrapper.find('.fa-sun').length).toBe(1);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<DaysCard day={dayObj} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
