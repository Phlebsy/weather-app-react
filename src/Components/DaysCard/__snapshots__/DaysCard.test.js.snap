// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DaysCard /> matches the snapshot 1`] = `
<div
  className="d-flex flex-column align-items-center py-3 bg-success"
>
  <i
    className="fas fa-sun"
  />
  <p
    className="m-0"
  >
    37
    Â°
  </p>
  <p
    className="m-0"
  >
    Tue - Dec 4
  </p>
</div>
`;
